programs/sbf/tests/sysvar.rs

```diff
@@ -40,6 +43,28 @@ fn test_sysvar_syscalls() {
         ..epoch_rewards::EpochRewards::default()
     };
     bank.set_sysvar_for_tests(&epoch_rewards);
+
+    // `StakeHistoryGetEntry` API assumes `MAX_ENTRIES` are present in the
+    // sysvar data.
+    //
+    // This isn't going to align with the bank's epoch, but it doesn't matter
+    // for the on-chain query, as long as there is data to retrieve.
```

Comment:
just to clarify, this code:
```rust
        let oldest_historical_epoch = current_epoch.saturating_sub(MAX_ENTRIES as u64);

        // target epoch is old enough to have fallen off history; presume fully active/deactive
        if target_epoch < oldest_historical_epoch {
            return None;
        }
```

doesnt assume there are `MAX_ENTRIES` in the data, it just uses the value to clamp the oldest epoch to 0. if stake history has fewer than 512 epochs of entries, it never reads before the first entry

Reply:
Ahh, I see. thanks! Looks like I misinterpreted the implementation. Updated my test!